src/js/components/bulkDownload/awards/filters/LocationFilter.jsx

```diff
@@ -0,0 +1,219 @@
+/**
+ * LocationFilter.jsx
+ * Created by Lizzie Salita 3/23/18
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import * as Icons from 'components/sharedComponents/icons/Icons';
+
+const propTypes = {
+    states: PropTypes.array,
+    currentLocation: PropTypes.object,
+    updateFilter: PropTypes.func
+};
+
+const countryOptions = [
+    {
+        code: 'USA',
+        name: 'United States'
+    },
+    {
+        code: 'FOREIGN',
+        name: 'All Foreign Countries'
+    },
+    {
+        code: '',
+        name: '---'
+    }
+];
+
+export default class LocationFilter extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showCountryPicker: false,
+            showStatePicker: false
+        };
+
+        this.toggleCountryPicker = this.toggleCountryPicker.bind(this);
+        this.handleCountrySelect = this.handleCountrySelect.bind(this);
+        this.toggleStatePicker = this.toggleStatePicker.bind(this);
+        this.handleStateSelect = this.handleStateSelect.bind(this);
+    }
+
+    toggleCountryPicker(e) {
+        e.preventDefault();
+        this.setState({
+            showCountryPicker: !this.state.showCountryPicker
+        });
+    }
+
+    handleCountrySelect(e) {
+        e.preventDefault();
+        const target = e.target;
+
+        this.props.updateFilter('location', {
+            country: {
+                code: target.value,
+                name: target.name
+            },
+            state: ''
+        });
+
+        this.setState({
+            showCountryPicker: false
+        });
+    }
+
+    toggleStatePicker(e) {
+        e.preventDefault();
+        if (this.props.currentLocation.country.code === 'USA') {
+            this.setState({
+                showStatePicker: !this.state.showStatePicker
+            });
+        }
+    }
+
+    handleStateSelect(e) {
+        e.preventDefault();
+        const target = e.target;
+
+        const updatedLocation = Object.assign({}, this.props.currentLocation, {
+            state: target.value
+        });
+
+        this.props.updateFilter('location', updatedLocation);
+
+        this.setState({
+            showStatePicker: false
+        });
+    }
+
+    render() {
+        const icon = (
+            <div className="icon valid">
+                <Icons.CheckCircle />
+            </div>
+        );
+
+        const countries = countryOptions.map((country) => (
+            <li
+                className="field-item indent"
+                key={`field-${country.code}`}>
+                <button
+                    className="item-button"
+                    title={country.name}
+                    aria-label={country.name}
+                    value={country.code}
+                    name={country.name}
```

Comment:
You should not be saving the country name in the `name` HTML attribute nor receiving it via the event `target.name` value. This is because `name` is intended to label the control's field when a form is sent to a server, not for holding a value. See https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes

Reply:
A more appropriate approach would be to store the code in the `value` attribute, and then look up the code to retrieve the associated data in your event handler.